### PR TITLE
feat: add video duration to video playback events

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR adds video duration to video playback events

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: models/video/video_playback_events.sql [hunk 1]
```diff
@@ -16,5 +16,6 @@ SELECT
     org,
     verb_id,
     ceil(CAST(coalesce(nullIf(JSON_VALUE(event, '$.result.extensions."https://w3id.org/xapi/video/extensions/time"'), ''), nullIf(JSON_VALUE(event, '$.result.extensions."https://w3id.org/xapi/video/extensions/time-from"'), ''), '0.0'), 'Decimal32(2)')) AS video_position
+    coalesce(nullIf(json_value(event, '$.context.extensions."https://w3id.org/xapi/video/extensions/length"'), ''), 0) as video_duration
```
Comment (bmtcril) [inline models/video/video_playback_events.sql:19]: I think you need a comma at the end of line 18

**Comment by bmtcril** [comment]
Can you split out different sets of changes into different PRs to make them easier to review?

**Review by SoryRawyer — APPROVED** [review]
In general, I think it'd be better to return a default value rather than returning nulls, in which case we could do something like `JSONExtractInt(event, 'context', 'extensions', 'https://w3id.org/xapi/video/extensions/length')` (which would default to 0 if that key is missing). That said, I don't think this should block the PR so feel free to include that change or ignore it.

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/aspects-dbt&entry.867055334=https://github.com/openedx/aspects-dbt/pull/46&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-02-27+17:20&entry.1917517419=2024-02-28+18:55&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**